sound/soc/sof/sof-ipc-test.c

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
+//
+#include <linux/module.h>
+#include <linux/virtual_bus.h>
+#include <linux/mod_devicetable.h>
+
+static int sof_ipc_test_probe(struct virtbus_device *vdev)
+{
+	dev_dbg(&vdev->dev, "probing %s\n", vdev->name);
+	return 0;
+}
+
+static int sof_ipc_test_remove(struct virtbus_device *vdev)
+{
+	return 0;
+}
+
+static void sof_ipc_test_shutdown(struct virtbus_device *vdev)
+{}
+
+static const struct virtbus_dev_id sof_ipc_virtbus_id_table[] = {
+	{"sof-ipc-test", 0},
+	{},
+};
+
+static struct virtbus_driver sof_ipc_test_vdrv = {
+	.driver = {
+		.name = "sof-ipc-test-drv",
+		.owner = THIS_MODULE,
+	},
+	.id_table = sof_ipc_virtbus_id_table,
+	.probe = sof_ipc_test_probe,
+	.remove = sof_ipc_test_remove,
+	.shutdown = sof_ipc_test_shutdown,
+};
+
+static int __init sof_ipc_test_init_module(void)
+{
+	int ret;
+
+	ret = virtbus_drv_register(&sof_ipc_test_vdrv);
+	if (ret) {
+		pr_err("Failed SOF IPC Test virtual driver register()\n");
+		return ret;
+	}
+
+	return 0;
+}
+
+static void __exit sof_ipc_test_exit_module(void)
+{
+	virtbus_drv_unregister(&sof_ipc_test_vdrv);
+}
+
+module_init(sof_ipc_test_init_module);
+module_exit(sof_ipc_test_exit_module);
```

Comment:
they probably need a module_virtbus_register as done for SoundWire slave and master.

Reply:
Yes, there are a few suggestions I have for the virtbus implementation but I am.unable to reach any of those guys.

Reply:
then send the suggestion as an RFC to the Intel drivers-review ML and CC: them